unity/unity-embed-host/CoreCLRHost.cs

```diff
@@ -607,6 +608,42 @@ public static IntPtr unity_method_get_function_pointer([NativeCallbackType("Mono
         return t.IsValueType;
     }
 
+    private static ConcurrentDictionary<Type, bool> s_isBlittableCache = new ConcurrentDictionary<Type, bool>();
+    [return: NativeCallbackType("gboolean")]
+    public static bool class_is_blittable(
+        [NativeCallbackType("MonoClass*")] IntPtr klass)
+    {
+        bool isBlittable = false;
+        Type t = klass.TypeFromHandleIntPtr();
+
+        if (s_isBlittableCache.TryGetValue(t, out isBlittable))
+        {
+            return isBlittable;
+        }
+
+        if (!t.IsValueType)
+        {
+            s_isBlittableCache[t] = false;
+            return false;
+        }
+
+        try
+        {
+            object tInstance = FormatterServices.GetUninitializedObject(t);
+            Type asmClass = Type.GetType("System.Runtime.InteropServices.Marshal");
```

Comment:
If we're caching this anyway to we want to just call `GCHandle.Alloc(tInstnace, GCHandleType.Pineed)` and catch the exception instead of calling an private method?

Reply:
I was going by what we ended up deciding when discussing at the office, but I don't mind changing to the pinned alloc try-catch if you prefer.

Reply:
As discussed offline, now using the alloc method.